lib/src/widgets/release_body.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:fwupd/fwupd.dart';
+import 'package:provider/provider.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+
+import '../../device_model.dart';
+import '../../fwupd_x.dart';
+import 'confirmation_dialog.dart';
+import 'release_card.dart';
+
+class ReleaseBody extends StatelessWidget {
+  const ReleaseBody({
+    super.key,
+    required this.device,
+    required this.releases,
+    required this.onInstall,
+  });
+
+  final FwupdDevice device;
+  final List<FwupdRelease> releases;
+  final ValueChanged<FwupdRelease> onInstall;
+
+  @override
+  Widget build(BuildContext context) {
+    final model = context.watch<DeviceModel>();
+    final selected = model.selectedRelease;
+    final l10n = AppLocalizations.of(context);
+    final String action;
+    final String dialogText;
+
+    if (selected?.isDowngrade == true) {
+      action = l10n.downgrade;
+      dialogText = l10n.downgradeConfirm(
+        device.name,
+        device.version,
+        selected?.version,
+      );
+    } else if (selected?.isUpgrade == false) {
+      action = l10n.reinstall;
+      dialogText = l10n.reinstallConfirm(
+        device.name,
+        device.version,
+      );
+    } else {
+      action = l10n.upgrade;
+      dialogText = l10n.upgradeConfirm(
+        device.name,
+        device.version,
+        selected?.version,
+      );
+    }
+
+    return Padding(
```

Comment:
hmm, i could be missing something but the structure looks very similar to `Scaffold` + `AppBar` + `ListView` + `ButtonBar`. is there a (stylistic?) reason to implement the page layout "manually" instead?

Reply:
No specific reason - your suggestion is better, I'll change it :)

Reply:
sorry for the bike-shedding, but perhaps we could even take it one step further to get rid of the column?
```dart
Scaffold(
  appBar: AppBar(),
  body: ListView(),
  bottomNavigationBar: ButtonBar(),
),
```

Reply:
ok, makes sense :+1: